ludwig/encoders/image_encoders.py

```diff
@@ -280,4 +282,71 @@ def __init__(
 
     def call(self, inputs, training=None, mask=None):
         hidden = self.mlp_mixer(inputs, training=training)
-        return {'encoder_output': hidden}
\ No newline at end of file
+        return {'encoder_output': hidden}
+
+
+@register(name='vit')
+class ViTEncoder(ImageEncoder):
+    def __init__(
+            self,
+            image_size,
+            patch_size: int = 16,
+            num_layers: int = 8,
+            d_model: int = 64,
+            num_heads: int = 4,
+            mlp_dim: int = 128,
+            channels: int = 3,
+            dropout: float = 0.1,
```

Comment:
To implement loading the weights, something you could do is have a param here called `use_pretrained_weights` or similar that, when true, will load weights into the layers. The weights can be either store in the repo if small, or we can consider serving them in s3 or similar if they're large.